### ISSUE TITLE
Transformer.Train per-sample SGD does not concentrate probability mass at V=256 byte-LM scale (reproduces on 0.179.0 + Tensors 0.71.0)

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1264]
## Summary

\`Transformer.Train(input, target)\` per-sample SGD path makes meaningful progress at small vocabularies (V=4) but **fails to escape unigram-prior accuracy at V=256 byte-LM** under any practical step budget. The byte-LM L-sweep on 100KB Shakespeare gets stuck at exactly 13.40% top-1 / PPL ~243 across L=1, 2, 4 and 1, 3 epochs — bit-identical numbers, which is the unigram floor (\`space\` is the most common byte). Reproduced on **the latest packages** (AiDotNet 0.179.0 + AiDotNet.Tensors 0.71.0).

## Standalone reproducer (pure AiDotNet, no third-party harness)

Single-file probe at \`scripts/aidotnet-convergence-probe/Program.cs\` in HarmonicEngine, project file pulls only the three AiDotNet packages. Runs in ~3s end-to-end. Output:

\`\`\`
AiDotNet convergence probe
  AiDotNet:         0.0.5.0
  AiDotNet.Tensors: 1.0.0.0
  Tensor<T> located in: AiDotNet.Tensors.LinearAlgebra.Tensor\`1

--- V=4, 200 steps, d=16, ff=32 ---
  P(target=1)     0.281084 ->   0.472721   (random ≈ 0.250000)
  logit[target]     0.3740 ->     0.9920
  L2(Δlogits): 0.737089
  Verdict:     P(target) > 0.5 after overfitting one example? NO

--- V=256, 1000 steps, d=32, ff=64 ---
  P(target=42)     0.003905 ->   0.010538   (random ≈ 0.003906)
  logit[target]     0.0037 ->     0.9991
  L2(Δlogits): 0.997625
  Verdict:     P(target) > 0.5 after overfitting one example? NO
\`\`\`

(Full repro scripts attached in repo — happy to inline if useful.)

## What's happening

Gradients flow correctly — \`logit[target]\` rises by **+0.6 (V=4) / +0.99 (V=256)** over the train budget. The bug is downstream:

- **At V=4**: After 200 steps, \`logit[target]\` ≈ 0.99 vs. others around -0.1, so softmax gives \`P(target) ≈ exp(0.99)/[exp(0.99) + 3·exp(-0.1)]\` ≈ 0.47. To cross 0.5 you'd need ~250-300 steps. So V=4 is **slowly** convergent.
- **At V=256**: After 1000 steps, \`logit[target]\` ≈ 1.0 vs. ~255 other logits near 0, so softmax gives \`P(target) ≈ exp(1)/[exp(1) + 255·exp(0)] ≈ 0.011\`. To cross 0.5 you'd need \`logit[target]\` ≥ \`ln(255) ≈ 5.5\` while others stay near 0. At the current per-step rate, that's ~5500 SGD steps **just to memorize ONE example**.

For real byte-LM training (1MB / 56k samples / 3 epochs), each pattern is seen only ~3 times — there's no chance the optimizer can lift the right \`logit[target]\` 5.5 units above noise. Hence the unigram floor.

## End-to-end byte-LM evidence

Same conclusion via the L-sweep test (\`tests/Learning/Phase_A4_TransformerLSweep_1MB.cs\` in HarmonicEngine):

\`\`\`
Phase A4 — AiDotNet Transformer L-sweep (V=256 ctx=64 d=64 ff=256 heads=4)
                                  3 epochs, 100KB Shakespeare, seed=0
  L         top1       top5        ppl       params     wall_s
  -----------------------------------------------------------------
  1       13.40%     41.80%     243.57       32,832       77.8
  2       13.40%     41.80%     243.83       49,280      151.8
  4       13.40%     41.80%     243.72       82,176      311.7
  -----------------------------------------------------------------
  Transformer L=1→2: +0.00pp
  Transformer L=2→4: +0.00pp
\`\`\`

13.40% top-1 = unigram (most-frequent byte = \`space\`). PPL ~243 ≈ \`exp(5.5)\` = effectively random over V=256. Identical at all 3 depths and 1 vs 3 epochs.

## Suspected root cause

Either:
1. The optimizer's effective step size is too small for cross-entropy at V=256 (e.g., LR not adjusted for output-class count).
2. The optimizer has no gradient accumulation — every \`Train()\` call applies a single-sample step in isolation, no momentum/Adam state across calls. Single-sample SGD is mathematically untenable at high V.
3. The loss gradient is being scaled by some factor (1/V?) that makes step size shrink with vocabulary.

## Suggested fixes (any one would be sufficient)

- **Mini-batch support**: \`Transformer.Train(Tensor<T>[B,S], Tensor<T>[B,V])\` accepting batched inputs would average gradients across B examples per step. Standard practice; PyTorch / TensorFlow only converge on byte-LM via batches of 32–256.
- **Adam-with-momentum default**: even per-sample, Adam smooths noise across thousands of steps and would close the gap on small-data tasks.
- **Documented learning-rate scaling**: if the framework expects users to bump LR with V, document it in the \`Transformer.Train\` XML doc.

## Workarounds available today

The \`AiModelBuilder\` facade with \`ConfigureOptimizer(new AdamOptimizer<...>())\` + \`ConfigureDataLoader\` is the documented path; the per-sample \`Transformer.Train\` looks like the lower-level escape hatch. We're hitting it directly because we want training to run sample-at-a-time for streaming corpus support, but mini-batches would be vastly better.

## Severity

Medium-high. The framework can be used productively via the facade. But \`Transformer.Train(input, target)\` is in the public API, no warning attached, and looks like the way to train. It silently no-ops at any V > 16 byte-LM scale. Documenting the limitation would be the absolute minimum.

## Versions tested

| Package | Version | Result |
|---|---|---|
| AiDotNet | 0.179.0 (latest) | reproduces |
| AiDotNet.Tensors | 0.71.0 (latest) | reproduces |
| AiDotNet.Native.OpenBLAS | 0.71.0 (latest) | reproduces |

Earlier 0.178.0 + 0.70.2: same.
